### PR TITLE
Cknieling/initial enum selection

### PR DESCRIPTION
**Pull request by chr11115**
- Add support for custom default values in enums

### CODE REVIEW

**Comment by chr11115** [comment]
Related PR, that is based on this PR:
https://github.com/Esri/serlio/pull/126

**Review by mistafunk — CHANGES_REQUESTED** [review]
Nice work! Found two issues...

File: src/serlio/modifiers/PRTModifierAction.cpp [hunk 1]
```diff
@@ -240,6 +240,74 @@ MStatus addHiddenBoolParameter(MFnDependencyNode& node, MFnAttribute& tAttr, con
...
+	}
+
+	MStatus status;
+	short idx = eAttr.fieldIndex(defMStringVal, &status);
+
+	const bool needToAddCustomDefaultValue = (status != MStatus::kSuccess);
+	const bool needToRemoveCustomDefaultValue = ((idx > 0) && (minVal == 0));
+
+	if (needToAddCustomDefaultValue || needToRemoveCustomDefaultValue) {
+		std::vector<MString> enumOptions;
```
[the remark this answers is not in the document]
Comment (chr11115) [reply src/serlio/modifiers/PRTModifierAction.cpp:286]: The array/vector is never directly fed to the Maya API and std::vector is a bit more readable to iterate over imo. However, if I would need to set the value of a stringArrayAttribute, I would definately use a MStringArray. Would you switch in this scenario?

[the remark this answers is not in the document]
Comment (chr11115) [reply src/serlio/modifiers/PRTModifierAction.cpp:286]: Not using MString anyways now..

File: src/serlio/modifiers/PRTModifierAction.cpp [hunk 2]
```diff
@@ -241,6 +240,74 @@ MStatus addHiddenBoolParameter(MFnDependencyNode& node, MFnAttribute& tAttr, con
...
+
+		for (short currIdx = 1; currIdx <= maxVal; currIdx++)
+			enumOptions.emplace_back(eAttr.fieldName(currIdx).asWChar());
+
+		MStatus stat;
+		const MFnDependencyNode fNode(node, &stat);
+		MCHECK(stat);
+		// clear enum options
+		MCHECK(mu::setEnumOptions(fNode.name().asWChar(), eAttr.name().asWChar(), {}));
```
[the remark this answers is not in the document]
Comment (chr11115) [reply src/serlio/modifiers/PRTModifierAction.cpp:295]: fixed in https://github.com/Esri/serlio/pull/126

**Review by mistafunk — APPROVED** [review]

**Pull request merged**